openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -66,25 +64,27 @@ def validate_instance(self, instance: Dict[str, Any]) -> None:
             raise ActionException(
                 "SingleSignOn field attributes are not configured in OpenSlides"
             )
-        additional_required_fields = {
-            value: self.model.saml_id.get_payload_schema()
-            for key, value in self.saml_attr_mapping.items()
-            if key == "saml_id"
-        }
-        additional_optional_fields = {
-            value: cast(Field, getattr(self.model, key, {})).get_payload_schema()
-            for key, value in self.saml_attr_mapping.items()
-            if key != "saml_id" and key in allowed_user_fields
-        }
         self.schema = {
             "$schema": schema_version,
             "title": "create saml account schema",
             "type": "object",
             "properties": {
-                **additional_required_fields,
-                **additional_optional_fields,
+                payload_field: {
+                    "oneOf": [
+                        (
+                            type_def := self.model.get_field(
+                                model_field
+                            ).get_payload_schema()
+                        ),
+                        {"type": "array", "items": type_def}
+                        if model_field != "saml_id"
+                        else {"type": "array", "items": type_def, "minItems": 1},
```

Comment:
```suggestion
                        {"type": "array", "items": type_def, "minItems": int(model_field == "saml_id")},
```
or, if that's too code-golfy:
```suggestion
                        {"type": "array", "items": type_def, "minItems": 1 if model_field == "saml_id" else 0},
```